frontend/src/app/main/data/workspace/transforms.cljs

```diff
@@ -260,6 +231,39 @@
 
     [root transformed-root ignore-geometry?]))
 
+(defn- set-modifiers-recursive
+  [modif-tree objects shape modifiers root transformed-root ignore-constraints]
+  (let [children (map (d/getf objects) (:shapes shape))
+
+        transformed-shape (gsh/transform-shape (assoc shape :modifiers modifiers))
+
+        [root transformed-root ignore-geometry?]
+        (check-delta shape root transformed-shape transformed-root objects)
+
+        modifiers (assoc modifiers :ignore-geometry? ignore-geometry?)
+
+        transformed-rect (gsh/calc-transformed-parent-rect shape modifiers)
+
+        set-child
+        (fn [modif-tree child]
+          (let [child-modifiers
+                (gsh/calc-child-modifiers shape child modifiers ignore-constraints transformed-rect)]
+
+            (cond-> modif-tree
+              (d/not-empty? (d/without-keys child-modifiers [:ignore-geometry?]))
+              (set-modifiers-recursive objects
+                                       child
+                                       child-modifiers
+                                       root
+                                       transformed-root
+                                       ignore-constraints))))
+
+        modif-tree
+        (-> modif-tree
+            (assoc-in [(:id shape) :modifiers] modifiers))]
```

Comment:
This is also minor, but a great tip when each `ns` of performance matters. All the `*-in` functions (assoc-in, update-in) are less performant because require create an ad-hoc vector for the path to modify.

In this case, this `assoc-in` can be implemented using `update` + `assoc`:

```clojure
(update modif-tree (:id shape) assoc :modifiers modifiers)
```

The `*-in` functions worth only when we have really a dynamic list (vector) of elements or the performance is not important. 

```
Execution time sample mean : 141.280365 ns (for assoc-in)
Execution time sample mean : 84.309110 ns (for update + assoc)
```

Reply:
Ok, I've found some others without-keys that don't make sense. I'll change those as well.